docs/topics/multiplatform/multiplatform-compatibility-guide.md

```diff
@@ -786,3 +786,70 @@ Here's the planned deprecation cycle:
 * \>2.0: remove the presets-related API from the public API of the Kotlin Gradle plugin; sources that still use it fail
   with "unresolved reference" errors, and binaries (for example, Gradle plugins) might fail with linkage errors
   unless recompiled against the latest versions of the Kotlin Gradle plugin
+
+## New approach to forward declarations
+
+**What's changed?**
+
+The JetBrains team has revamped the approach to forward declarations in Kotlin to make their behavior more predictable
+and prepare this functionality for the upcoming Kotlin 2.0 release. From now on:
+
+* You can only import forward declarations using the `cnames` or ` objcnames` packages.
+* You need to explicitly make a cast to and from the corresponding C and Objective-C forward declaration.
+
+**What's the best practice now?**
+
+* Consider a C library with a `library.package` that declares a `cstructName` forward declaration.
+  Previously, it was possible to import it directly from the library with `import library.package.cstructName`.
+  Now, you can only use a special forward declaration package for that: `import cnames.structs.cstructName`.
+  The same is true for `objcnames`.
+
+*  Consider two cinterop libraries, one that uses `cnames.structs.cstructName` and the other that has an actual definition:
```

Comment:
`cnames.struct.cstructName` is mentioned here and after the source code fragments. But the source code fragments use `objcnames.protocols.ForwardDeclaredProtocolProtocol` instead.